### PR TITLE
Wiki integration & some bug fixes

### LINKED ISSUE #70: RSC wiki integration
Similar to the integrations in the RS3 and OSRS clients, it would be really nice to have a button somewhere in the interface that you could click to toggle click opening the relevant wiki page.

Maybe underneath the inventory satchel would be an unobtrusive place.

### LINKED ISSUE #86: Option to hide "Report abuse" button
A few hooks needed for this I think.

Some are in
```java
    private void drawChatMessageTabs(boolean var1) {
```

First need a hook to disable `this.surface.drawSprite((byte) 61, -4 + this.gameHeight, this.spriteMedia + 23, 0);`. 

At the same place, need a hook to draw our own image there instead. 

Then we need a hook to conditionally disable drawing 
`this.surface.drawstringCenter("Report abuse", 0, this.gameHeight - -6, 457, 0xffffff);`

It would be additionally nice if we could inject our own logic to drawstringCenter afterwards, to facilitate #70.

Finally, where this code is
```java
                if (super.mouseX > 417 && super.mouseX < 497 && super.lastMouseButtonDown == 1) {
                    this.showDialogReportAbuseStep = 1;
                    super.inputTextCurrent = "";
                    super.inputTextFinal = "";
                }
```
we need to abort that functionality, and add a hook to call our own function instead (which could be Nothing, if button is completely disabled, but could be #70)

It is worth noting that even if this "Report abuse" button is disabled, the ability to report players will remain accessible via the right click menu of players.

### PR DESCRIPTION
**Pull request by Hubcapp**
Wiki Integration! Cool!

Here's a video of an older implementation than is being PR'd here, it shows the basic idea though.

https://cdn.discordapp.com/attachments/651377067686101032/830251227988230164/rscplus.wiki.integration.mp4

~~I would really like #86 complete in order to have space on screen for another button. For now, I've only implemented wiki support as an opt-in feature of clicking on the magic book. It has to be opt-in until #86 is done, because it is too easy to misclick the book while spellcasting. I imagine the "clicking on the magic book" option will remain available, even after #86, but replacing the report button  with a wiki-lookup button (if there is not enough space to fit both) will be the default option in RSC+ once possible.~~

This PR also fixes the following bugs:
* Users have been confused on how to sign up for a new account using RSC+. We will now always show the first screen at least the first time the client is run, so that users know the screen exists.
* Zoom & rotation do not actually authentically get reset every time the user logs in. I have changed it so that Zoom will be reset when using inauthentic zoom, but otherwise, the client will behave authentically. (also no longer reset bank active page or combat timer)
* Fix bug where Friends' hitboxes (ctrl-h) were not highlighted green unless overhead names were also enabled
* Last pixel of the hbar authentically has 1 pixel missing at the end, and now fixing that is on a toggle (default off)
* 5px vertical spacing bug in wrench menu did not have a toggle

resolves #70
resolves #86

(Also I decided to reorganize the Bank interface assets into their own folder & add a new "don't add inventory items" mode)

### CODE REVIEW

**Review by lmsv-mx123 — COMMENTED** [review]
Have looked at the code and overall looks good, probably will test in a bit

**Comment by lmsv-mx123** [comment]
One consideration probably would be good is for the wiki to check if the user can access the internet (or even the wiki in case of firewall or whatsoever), otherwise we could avoid them the trouble of opening the browser and fail instead

**Comment by Hubcapp** [comment]
> One consideration probably would be good is for the wiki to check if the user can access the internet (or even the wiki in case of firewall or whatsoever), otherwise we could avoid them the trouble of opening the browser and fail instead

Probably more trouble than worth. Don't really want to add network detection code to RSC+ (would necessitate needlessly trying to wget from the wiki on every request). They can just have the browser tell them they are offline if they try to look something up on the wiki and can't.

**Comment by lmsv-mx123** [comment]
I was thinking more of the lines of something like this: https://stackoverflow.com/questions/1402005/how-to-check-if-internet-connection-is-present-in-java

**Comment by Hubcapp** [comment]
> I was thinking more of the lines of something like this: https://stackoverflow.com/questions/1402005/how-to-check-if-internet-connection-is-present-in-java

It's like the same thing. The specific case you mention of network firewall, you would need to check on port 443 with the https protocol, if the network will let it pass the traffic or not. Pinging wiki won't reveal if it will work or not (wiki may choose to block pings at some point, and also may be weird for them to see in their traffic logs). We will turn RSC+ into a web browser minus the renderer if we do this.

Basically I think it is fine to trust the user that if they activate the wiki lookup tool, they intend to look something up on the wiki. and we can let them find out if it works or not. Even if it doesn't work, they will have the link open in their browser for the next time network connectivity becomes available.

**Comment by lmsv-mx123** [comment]
> > I was thinking more of the lines of something like this: https://stackoverflow.com/questions/1402005/how-to-check-if-internet-connection-is-present-in-java
> 
> It's like the same thing. The specific case you mention of network firewall, you would need to check on port 443 with the https protocol, if the network will let it pass the traffic or not. Pinging wiki won't reveal if it will work or not (wiki may choose to block pings at some point, and also may be weird for them to see in their traffic logs). We will turn RSC+ into a web browser minus the renderer if we do this.
> 
> Basically I think it is fine to trust the user that if they activate the wiki lookup tool, they intend to look something up on the wiki. and we can let them find out if it works or not. Even if it doesn't work, they will have the link open in their browser for the next time network connectivity becomes available.

Yea I guess maybe was overthinking of users, most are that should be connected to internet either way to play

**Comment by Hubcapp** [comment]
I've watched replays on a plane before without internet. I think even in the situation without internet, if I wanted to look something up, I would want the web browser to open the links anyway, so that I can come back to that thought later.

**Comment by lmsv-mx123** [comment]
That's true, then it's fine
